src/commands/wp.ts

```diff
@@ -0,0 +1,255 @@
+/**
+ * External dependencies
+ */
+import chalk from 'chalk';
+import debugLib from 'debug';
+import gql from 'graphql-tag';
+import * as ssh2 from 'ssh2';
+
+/**
+ * Internal dependencies
+ */
+import pkg from '../../package.json';
+import { App, AppEnvironment } from '../graphqlTypes';
+import API from '../lib/api';
+import { CommandTracker, makeCommandTracker } from '../lib/tracker';
+
+const debug = debugLib( '@automattic/vip:wp/ssh' );
+
+const NON_TTY_COLUMNS = 100;
+const NON_TTY_ROWS = 15;
+
+const SSH_HANDSHAKE_TIMEOUT_MS = 5000;
+
+interface TriggerWPCLICommandMutationResponse {
+	triggerWPCLICommandOnAppEnvironment: {
+		inputToken: string;
+		sshAuthentication: {
+			host: string;
+			port: number;
+			username: string;
+			privateKey: string;
+			passphrase: string;
+		};
+		command: {
+			guid: string;
+		};
+	};
+}
+
+const TRIGGER_WP_CLI_COMMAND_MUTATION = gql`
+	mutation TriggerWPCLICommandMutation($input: AppEnvironmentTriggerWPCLICommandInput) {
+		triggerWPCLICommandOnAppEnvironment(input: $input) {
+			inputToken
+			sshAuthentication {
+				host
+				port
+				username
+				privateKey
+				passphrase
+			}
+			command {
+				guid
+			}
+		}
+	}
+`;
+
+const getSSHAuthForCommand = async ( appId: number, envId: number, command: string ) => {
+	const api = API();
+
+	return api.mutate< TriggerWPCLICommandMutationResponse >( {
+		mutation: TRIGGER_WP_CLI_COMMAND_MUTATION,
+		variables: {
+			input: {
+				id: appId,
+				environmentId: envId,
+				command,
+			},
+		},
+	} );
+};
+
+export class NonZeroExitCodeError extends Error {
+	public exitCode: number;
+
+	constructor( message: string, exitCode: number ) {
+		super( message );
+		this.exitCode = exitCode;
+	}
+}
+
+export class WPCliCommandOverSSH {
+	private app: App;
+	private env: AppEnvironment;
+	private track: CommandTracker;
+
+	constructor( app: App, env: AppEnvironment ) {
+		this.app = app;
+		this.env = env;
+
+		this.track = makeCommandTracker( 'wp', {
+			app: this.app.id,
+			env: this.env.id,
+			execution_type: 'ssh',
+		} );
+	}
+
+	public async run( command: string ): Promise< void > {
+		if ( ! this.app.id || ! this.env.id ) {
+			await this.track( 'error', {
+				error: 'no_app_env_id',
+				message: 'No app or env ID provided',
```

Comment:
In `src/bin/vip-wp.js`, we also include a stripped down version of the command. Should we do that here as well? This can be done in a separate PR.